static/main.js

```diff
@@ -111,46 +111,64 @@ var redraw_counter      = 0;
 var history_position    = 0;
 
 //===================================
-//  Utility Functions
+//  Local Cache Functions
 //===================================
 
-function setCookie(cname, cvalue, exdays)
+//sets localStorage when value is changed
+function setCache(cname, cvalue)
 {
-    var d = new Date();
-    d.setTime(d.getTime() + (exdays*24*60*60*1000));
-    var expires = "expires="+ d.toUTCString();
-    document.cookie = cname + "=" + cvalue + ";" + expires + ";path=/";
+  localStorage.setItem(cname, JSON.stringify(cvalue));
 }
 
-function getCookie(cname)
+function checkCache(cname)
 {
-    var name = cname + "=";
-    var decodedCookie = decodeURIComponent(document.cookie);
-    var ca = decodedCookie.split(';');
-    for(var i = 0; i <ca.length; i++)
-    {
-        var c = ca[i];
-        while (c.charAt(0) == ' ')
-        {
-            c = c.substring(1);
-        }
-        if (c.indexOf(name) == 0)
-        {
-            return c.substring(name.length, c.length);
-        }
-    }
-    return "";
+  if (localStorage.getItem(cname) != null)
+  {
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
+
+function getCache(cname)
+{
+  return JSON.parse(localStorage.getItem(cname));
 }
 
-function checkCookie(cookie)
+function getCommandCache()
 {
-    var user = getCookie(cookie);
-    var result = false;
-    if (user != "")
+  if (localStorage.getItem('command_history') != null)
+  {
+    var get_storage = JSON.parse(localStorage.getItem('command_history'));
+    command_history = get_storage;
+    for(var i = 0; i < command_history.length; i++)
     {
-        result = true;
+        past_commands += '<option value="'+command_history[i]+'" />';
     }
-    return result;
+    document.getElementById('command-history').innerHTML = past_commands;
+    console.info("COMMAND CACHE RETRIEVED");
+  }
+
+  else
+  {
+    command_history = [];
+    console.info("NO COMMAND CACHE");
+  }
+}
+
+window.onbeforeunload => ()
```

Comment:
```suggestion
window.onbeforeunload = () =>
```
This is not proper syntax. It should be what I have put above. You `() =>` is the declaration of a function that you are assigning to window.onbeforeunload.